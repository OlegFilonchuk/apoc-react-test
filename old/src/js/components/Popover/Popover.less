@import 'PopoverVariables';
@import 'PopoverMixins';

@sizeBetweenContentAndPopover: 20px;

.tooltip-container {
  position: relative;
  display: inline-block;

  .tooltip-body {
    display: inline-block;
  }

  .popover-title {
    cursor: default;
  }

  .qtip {
    z-index: @qtip-zindex;
    min-width: @qtip-min-width;

    .qtip-close {
      display: none;
    }

    &.position-fixed {
      position: fixed;
    }

    &.qtip-is-closable .qtip-close {
      display: block;
    }

    &::before,
    &:after {
      right: @full-size;
      top: @negative-full-size;
      border: solid transparent;
      content: ' ';
      height: 0;
      width: 0;
      position: absolute;
      pointer-events: none;
    }

    &.qtip-chameleon-default {
      &::before {
        .qtip-arrow(@qtip-blue-border-color, @enabledBorderColor, @qtip-arrow-dark-border-width, 0, 0);
      }

      &::after {
        .qtip-arrow(@qtip-light-blue-border-color, @enabledBgColor, @qtip-arrow-light-border-width, 0, 1px);
      }
    }

    &.sc-qtip-chameleon-warning {
      &::before {
        .qtip-arrow(@qtip-yellow-border-color, @warningBorderColor, @qtip-arrow-dark-border-width, 0, 0);
      }

      &::after {
        .qtip-arrow(@qtip-light-yellow-border-color, @warningBgColor, @qtip-arrow-light-border-width, 0, 1px);
      }

      header h2 {
        border-bottom: 1px solid @warningBorderColor;
      }
    }

    &.sc-qtip-chameleon-error {
      &::before {
        .qtip-arrow(@qtip-red-border-color, @errorBorderColor, @qtip-arrow-dark-border-width, 0, 0);
      }

      &::after {
        .qtip-arrow(@qtip-light-red-border-color, @errorBgColor, @qtip-arrow-light-border-width, 0, 1px);
      }

      header h2 {
        border-bottom: 1px solid @errorBorderColor;
      }
    }

    &.qtip-without-arrow {
      &::before,
      &:after {
        display: none;
      }

      &.qtip-pos-lt {
        left: 100%;
        top: calc(100% - 32px);
      }

      &.qtip-pos-rt {
        right: 100%;
        top: calc(100% - 32px);
      }

      &.qtip-pos-tr {
        left: 100%;
        top: 100%;
      }

      &.qtip-pos-tl {
        right: 100%;
        top: 100%;
      }
    }

    &.qtip-disabled {
      border: @one-pixel-size solid @disabledBorderColor;
      background-color: @disabledBgColor;

      &::before {
        .qtip-arrow(@qtip-blue-border-color, @disabledBorderColor, @qtip-arrow-dark-border-width, 0, 0);
      }

      &::after {
        .qtip-arrow(@qtip-light-blue-border-color, @disabledBgColor, @qtip-arrow-light-border-width, 0, 1px);
      }
    }

    &.qtip-pos-lt {
      .position-qtip(@half-size, @full-width-plus-anchor-width-plus-empty-space);

      &::before,
      &::after {
        right: @full-size;
        top: @negative-one-pixel;
      }
    }

    &.qtip-pos-rt {
      .position-qtip(@half-size, @qtip-pos-rt-left);
      right: @full-width-plus-anchor-width-plus-empty-space;

      &::before,
      &::after {
        .position-qtip(@negative-one-pixel, @full-size, @flip-horizontaly);
      }

      &::after {
        border-top: 0;
        margin-top: @one-pixel-size;
      }
    }

    &.qtip-pos-tr {
      .build-qtip-pos-styles(
        @full-width-plus-anchor-width,
        @half-size,
        @qtip-pos-tr-transform,
        @qtip-tr-tl-arrow-top,
        @qtip-pos-tr-arrow-left,
        @rotate-by-90deg-clockwise,
        @anchorWidth,
        @qtip-tr-tl-margin-top,
        @one-pixel-size
      );
    }

    &.qtip-pos-tl {
      .build-qtip-pos-styles(
        @full-width-plus-anchor-width,
        @half-size,
        @default-transform-value,
        @qtip-tr-tl-arrow-top,
        @qtip-pos-tl-arrow-left,
        @qtip-pos-tl-arrow-transform,
        @qtip-arrow-dark-border-width,
        @qtip-tr-tl-margin-top,
        @one-pixel-size
      );
    }

    &.qtip-pos-br {
      .build-qtip-pos-styles(
        @negative-full-size,
        @half-size,
        @qtip-pos-br-transform,
        @full-size,
        @qtip-pos-br-arrow-left,
        @default-transform-value,
        @qtip-arrow-dark-border-width,
        @qtip-br-bl-margin-top,
        @negative-one-pixel
      );
    }

    &.qtip-pos-bl {
      .build-qtip-pos-styles(
        @negative-full-size,
        @half-size,
        @qtip-pos-bl-transform,
        @full-size,
        @negative-one-pixel,
        @qtip-pos-bl-arrow-transform,
        @qtip-arrow-dark-border-width,
        @qtip-br-bl-margin-top,
        @one-pixel-size
      );
    }

    &.qtip-pos-tm {
      .build-qtip-pos-styles(
        @full-width-plus-anchor-width,
        @initial-size,
        @default-transform-value,
        @qtip-tm-arrow-top,
        @qtip-pos-tm-arrow-left,
        @qtip-pos-tm-arrow-transform,
        @qtip-arrow-dark-border-width,
        @one-pixel-size,
        @initial-size
      );
    }

    &.qtip-pos-rm {
      .build-qtip-pos-styles(
        @qtip-pos-rm-lm-top,
        @negative-full-size-minus-double-anchor-width,
        @default-transform-value,
        @half-size,
        @qtip-pos-rm-arrow-left,
        @qtip-pos-rm-arrow-transform,
        @qtip-arrow-dark-border-width,
        @initial-size,
        @negative-one-pixel
      );
    }

    &.qtip-pos-lm {
      .build-qtip-pos-styles(
        @qtip-pos-rm-lm-top,
        @full-size-plus-double-anchor-width,
        @default-transform-value,
        @qtip-lm-arrow-top,
        @qtip-pos-lm-arrow-left,
        @qtip-pos-lm-arrow-transform,
        @qtip-arrow-dark-border-width,
        @initial-size,
        @one-pixel-size
      );
    }

    &.qtip-pos-bm {
      .build-qtip-pos-styles(
        @negative-full-size,
        @initial-size,
        @qtip-pos-bl-transform,
        @qtip-bm-arrow-top,
        @qtip-pos-bm-arrow-left,
        @qtip-pos-bm-arrow-transform,
        @qtip-arrow-dark-border-width,
        @negative-one-pixel,
        @initial-size
      );
    }

    .qtip-content {
      /**
        These rules doesn't change how component looks, but make possibility
        to move cursor from content to popover without Popover's props `hideAfterDelayOf`
      */
      border-top: @sizeBetweenContentAndPopover solid transparent;
      margin-top: -@sizeBetweenContentAndPopover;

      .visible {
        display: block;
        visibility: visible;

        header a:hover {
          text-decoration: none;
        }
      }
    }
  }
}
