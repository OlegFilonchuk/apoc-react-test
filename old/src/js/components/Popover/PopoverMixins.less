.position-qtip(@top, @left) {
  left: @left;
  top: @top;
}

.position-qtip(@top, @left, @transform) {
  .position-qtip(@top, @left);

  transform: @transform;
}

.qtip-arrow(@borderWidth, @marginTop, @marginLeft) {
  border-width: @borderWidth;
  margin-top: @marginTop;
  margin-left: @marginLeft;
}

.qtip-arrow(@borderColor, @borderRightColor, @borderWidth, @borderTop, @marginTop) {
  border-color: @borderColor;
  border-right-color: @borderRightColor;
  border-width: @borderWidth;
  border-top: @borderTop;
  margin-top: @marginTop;
}

.build-qtip-pos-styles(
  @qtip-pos-top, @qtip-pos-left, @qtip-pos-transform,
  @qtip-pos-arrow-top, @qtip-pos-arrow-left, @qtip-pos-arrow-transform,
  @qtip-pos-arrow-border-width, @qtip-pos-arrow-margin-top, @qtip-pos-arrow-margin-left
) {
  .position-qtip(@qtip-pos-top, @qtip-pos-left, @qtip-pos-transform);

  &::before,
  &::after {
    .position-qtip(@qtip-pos-arrow-top, @qtip-pos-arrow-left, @qtip-pos-arrow-transform);
  }

  &::after {
    .qtip-arrow(@qtip-pos-arrow-border-width, @qtip-pos-arrow-margin-top, @qtip-pos-arrow-margin-left);
  }
}
