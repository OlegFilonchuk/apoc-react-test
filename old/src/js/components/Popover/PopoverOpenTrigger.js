const POPOVER_OPEN_TRIGGER = {
  MOUSE_OVER: 'mouseOver',
  MOUSE_CLICK: 'mouseClick',
  MANUAL: 'manual',
  CLICK_OUTSIDE: 'click_outside'
};

Object.freeze(POPOVER_OPEN_TRIGGER);

export default POPOVER_OPEN_TRIGGER;
