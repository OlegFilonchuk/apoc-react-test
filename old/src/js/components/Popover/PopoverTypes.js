const POPOVER_TYPES = {
  WARNING: 'warning',
  ERROR: 'error',
  DEFAULT: 'default'
};

Object.freeze(POPOVER_TYPES);

export default POPOVER_TYPES;
