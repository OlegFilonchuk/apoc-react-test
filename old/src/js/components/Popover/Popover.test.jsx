import React from 'react';
import { shallow, mount } from 'enzyme';

import Popover from './Popover';
import POPOVER_PLACEMENTS from './PopoverPlacements';
import POPOVER_OPEN_TRIGGER from './PopoverOpenTrigger';

describe('<Popover />', () => {
  const DEFAULT_DELAY = 500;

  describe('default bevahior', () => {
    it(`should have default title and placement = ${Object.keys(POPOVER_PLACEMENTS)[0]}`, () => {
      const popoverContent = 'Some content for popover';
      const popover = shallow(<Popover>{popoverContent}</Popover>);

      expect(popover).toMatchSnapshot();
    });
  });

  describe('rendering with', () => {
    describe('default placement and', () => {
      describe('simple text', () => {
        it('should match snapshot', () => {
          const simpleTextTitle = 'Simple Text Title';
          const simpleTextContent = 'Simple Popover Text Content';
          const popover = shallow(<Popover title={simpleTextTitle}>{simpleTextContent}</Popover>);

          expect(popover).toMatchSnapshot();
        });
      });

      describe('React.Element as a title', () => {
        it('should match snapshot', () => {
          const Button = () => <button>Simple React.Element</button>;
          const popover = shallow(<Popover title={<Button />}>Simple Popover Text Content</Popover>);

          expect(popover).toMatchSnapshot();
        });
      });
    });

    describe('default title and', () => {
      Object.keys(POPOVER_PLACEMENTS).forEach(placement => {
        describe(`placement="${POPOVER_PLACEMENTS[placement]}"`, () => {
          it('should match snapshot', () => {
            const popover = shallow(
              <Popover placement={POPOVER_PLACEMENTS[placement]}>Simple Popover Text Content</Popover>
            );

            expect(popover).toMatchSnapshot();
          });
        });
      });
    });
  });

  describe('open/close logic', () => {
    describe('when `isVisible` property is `true`', () => {
      describe('and `openTrigger` is manual', () => {
        it('should show popover', () => {
          const popover = mount(<Popover isVisible openTrigger={POPOVER_OPEN_TRIGGER.MANUAL} />);

          expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');
        });
      });
    });

    describe('when `isAllowed` property is', () => {
      describe('false', () => {
        it('should NOT show popover and not react to mouse events', () => {
          const popover = mount(<Popover isAllowed={false} />);

          popover.find('.popover-title').simulate('mouseover');

          expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');

          popover.find('.qtip').simulate('mouseleave');

          expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');
        });
      });

      describe('true', () => {
        describe('and user moves mouse over `.popover-title`', () => {
          describe('and then mouse leaves `.qtip`', () => {
            it('should show popover and then hide immediately', () => {
              const popover = mount(<Popover isAllowed />);

              popover.find('.popover-title').simulate('mouseover');

              expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');

              popover.find('.qtip').simulate('mouseleave');

              expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');
            });
          });

          describe('and then mouse leaves `.popover-title`', () => {
            it(`should show popover and hide after delay of ${DEFAULT_DELAY}`, () => {
              const popover = mount(<Popover isAllowed />);

              popover.find('.popover-title').simulate('mouseover');

              expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');

              popover.find('.popover-title').simulate('mouseleave');

              return new Promise(resolve => {
                setTimeout(() => {
                  resolve();
                }, DEFAULT_DELAY);
              }).then(() => {
                expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');
              });
            });
          });
        });
      });
    });

    describe('when custom timer for hiding was not specified', () => {
      it(`should hide Popover after ${DEFAULT_DELAY}`, () => {
        const popover = mount(<Popover isAllowed />);

        popover.find('.popover-title').simulate('mouseover');
        popover.find('.popover-title').simulate('mouseleave');

        return new Promise(resolve => {
          setTimeout(() => {
            resolve();
          }, DEFAULT_DELAY);
        }).then(() => {
          expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');
        });
      });
    });

    describe('when user clicks close button', () => {
      it('should close popover immediately', () => {
        const popover = mount(<Popover isClosable>Some test content</Popover>);

        popover.find('.popover-title').simulate('mouseover');

        expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');

        popover
          .find('.qtip')
          .find('a')
          .simulate('click');

        expect(popover.find('.qtip').getDOMNode().style.display).toEqual('none');
      });
    });

    describe('when `openTrigger` property is specified and user', () => {
      describe('hovers `.popover-title`', () => {
        it('should NOT open Popover', () => {
          const popover = mount(<Popover openTrigger={POPOVER_OPEN_TRIGGER.CLICK_OUTSIDE}>Some test content</Popover>);

          const popoverTitle = popover.find('.popover-title');

          popoverTitle.simulate('mouseover');

          expect(popoverTitle).not.toHaveClassName('popover-title--visible');
          expect(popover.find('.qtip').getDOMNode().style.display).not.toEqual('block');
        });
      });

      describe('clicks `.popover-title`', () => {
        it('should add `popover-title--visible` and open Popover', () => {
          const popover = mount(<Popover openTrigger={POPOVER_OPEN_TRIGGER.CLICK_OUTSIDE}>Some test content</Popover>);

          const popoverTitle = popover.find('.popover-title');

          popoverTitle.simulate('click');

          expect(popoverTitle.getDOMNode().className).toContain('popover-title--visible');
        });
      });
    });

    describe('when `showAfterDelayOf` property is specified', () => {
      it('should show Popover after specified delay', () => {
        const popoverContent = 'Some content for popover';
        const showAfterDelayOf = 550;

        const popover = mount(<Popover showAfterDelayOf={showAfterDelayOf}>{popoverContent}</Popover>);

        const popoverTitle = popover.find('.popover-title');
        const qtip = popover.find('.qtip');

        expect(qtip.getDOMNode().style.display).toEqual('none');

        popoverTitle.simulate('mouseover');

        expect(qtip.getDOMNode().style.display).toEqual('none');

        return new Promise(resolve => {
          setTimeout(() => {
            resolve();
          }, showAfterDelayOf);
        }).then(() => {
          expect(qtip.getDOMNode().style.display).toEqual('block');
        });
      });
    });
  });

  describe('delays testing', () => {
    describe('when timer has already been set', () => {
      it('should not set timer again', () => {
        const LESS_THAN_DEFAULT_DELAY = 200;

        const popover = mount(<Popover isAllowed />);

        popover.find('.popover-title').simulate('mouseover');

        expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');

        popover.find('.popover-title').simulate('mouseleave');

        return new Promise(resolve => {
          setTimeout(() => {
            resolve();
          }, LESS_THAN_DEFAULT_DELAY);
        }).then(() => {
          expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block');
        });
      });
    });

    describe('when user hovers `.qtip` element', () => {
      it('should clear timeout and not hide `Popover`', () => {
        const popover = mount(<Popover isAllowed />);

        popover.find('.popover-title').simulate('mouseover');
        popover.find('.popover-title').simulate('mouseleave');
        popover.find('.qtip').simulate('mouseover');

        return new Promise(resolve => {
          setTimeout(() => {
            resolve();
          }, DEFAULT_DELAY);
        }).then(() => expect(popover.find('.qtip').getDOMNode().style.display).toEqual('block'));
      });
    });
  });

  describe('when `popoverStyles` property is specified', () => {
    it('should apply custom styles', () => {
      const additionalStyles = {
        color: 'green',
        border: '1px solid blue'
      };
      const popover = shallow(<Popover isAllowed popoverStyles={additionalStyles} />);

      expect(popover).toMatchSnapshot();
    });
  });

  describe('when Popover is closed, `onClose` prop', () => {
    it('should be called', () => {
      const onCloseSpy = jest.fn();

      const popover = mount(<Popover onClose={onCloseSpy} />);

      popover.find('.popover-title').simulate('mouseover');

      popover.find('.qtip').simulate('mouseleave');

      expect(onCloseSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('when `removeChildrenFromDOMWhenHidden` prop set to', () => {
    describe('`true` and Popover is hidden', () => {
      it('should not render children', () => {
        const wrapper = mount(
          <Popover removeChildrenFromDOMWhenHidden>
            <div />
          </Popover>
        );

        expect(wrapper.find('.tooltip-body > div')).not.toBePresent();
      });
    });

    describe('`true` and Popover is visible', () => {
      it('should render children', () => {
        const wrapper = mount(
          <Popover removeChildrenFromDOMWhenHidden>
            <div />
          </Popover>
        );

        wrapper.find('.popover-title').simulate('mouseover');

        expect(wrapper.find('.tooltip-body > div')).toBePresent();
      });
    });

    describe('`false` and Popover is hidden', () => {
      it('should render children', () => {
        const wrapper = mount(
          <Popover removeChildrenFromDOMWhenHidden={false}>
            <div />
          </Popover>
        );

        expect(wrapper.find('.tooltip-body > div')).toBePresent();
      });
    });
  });

  describe('when `followCursorPosition` prop is set to `true', () => {
    it('should add class `position-fixed` to qtip component', () => {
      const popover = shallow(<Popover placement={POPOVER_PLACEMENTS.BOTTOM_LEFT} followCursorPosition />);

      expect(popover.find('.qtip')).toHaveClassName('position-fixed');
    });
  });

  // TODO: since for now it's not possible to simulate `mousemove` event on a
  // TODO: component I mark this tests group as skipped until it will be fixed on enzyme-react
  describe.skip('when `followCursorPosition` prop is set to `true', () => {
    const MOUSEOVER_EVENT = 'mouseover';
    const MOUSEMOVE_EVENT = 'mousemove';
    const MOUSELEAVE_EVENT = 'mouseleave';

    describe('and `title component`', () => {
      describe('is `hovered`', () => {
        const addBackup = { ...() => {}, ...Node.prototype.addEventListener };
        const removeBackup = { ...() => {}, ...Node.prototype.removeEventListener };

        beforeEach(() => {
          Node.prototype.addEventListener = jest.fn();
          Node.prototype.removeEventListener = jest.fn();
        });

        afterEach(() => {
          Node.prototype.addEventListener = addBackup;
          Node.prototype.removeEventListener = removeBackup;
        });

        describe('when `placement`', () => {
          describe(`equal to "${POPOVER_PLACEMENTS.BOTTOM_LEFT}"`, () => {
            it(`should add "${MOUSEMOVE_EVENT}" event listener`, () => {
              const popover = mount(<Popover placement={POPOVER_PLACEMENTS.BOTTOM_LEFT} followCursorPosition />);
              const titleComponent = popover.find('.popover-title');

              titleComponent.simulate(MOUSEOVER_EVENT);

              expect(Node.prototype.addEventListener).toHaveBeenCalledWith(MOUSEMOVE_EVENT, jasmine.anything());
            });
          });

          describe(`equal to "${POPOVER_PLACEMENTS.BOTTOM_RIGHT}"`, () => {
            it(`should add "${MOUSEMOVE_EVENT}" event listener`, () => {
              const popover = mount(<Popover placement={POPOVER_PLACEMENTS.BOTTOM_RIGHT} followCursorPosition />);
              const titleComponent = popover.find('.popover-title');

              titleComponent.simulate(MOUSEOVER_EVENT);

              expect(Node.prototype.addEventListener).toHaveBeenCalledWith(MOUSEMOVE_EVENT, jasmine.anything());

              titleComponent.simulate(MOUSEMOVE_EVENT, { clientX: 0, clientY: 0 });
            });
          });

          describe(`is NOT equal to both "${POPOVER_PLACEMENTS.BOTTOM_LEFT}" or "${POPOVER_PLACEMENTS.BOTTOM_RIGHT}"`, () => {
            const placments = Object.keys(POPOVER_PLACEMENTS).map(placement => POPOVER_PLACEMENTS[placement]);

            placments.forEach(placement => {
              if (placement === POPOVER_PLACEMENTS.BOTTOM_LEFT || placement === POPOVER_PLACEMENTS.BOTTOM_RIGHT) {
                return;
              }

              describe(`and equal to ${placement}`, () => {
                it(`should not add ${MOUSEMOVE_EVENT} event listener`, () => {
                  const popover = mount(<Popover placement={placement} followCursorPosition />);
                  const titleComponent = popover.find('.popover-title');

                  titleComponent.simulate(MOUSEOVER_EVENT);

                  expect(Node.prototype.addEventListener).not.toHaveBeenCalledWith(MOUSEMOVE_EVENT, jasmine.anything());
                });
              });
            });
          });
        });

        describe(`and on "${MOUSELEAVE_EVENT}"`, () => {
          it(`should remove "${MOUSEMOVE_EVENT}" listener from "title component"`, () => {
            const popover = mount(<Popover placement={POPOVER_PLACEMENTS.BOTTOM_LEFT} followCursorPosition />);

            const titleComponent = popover.find('.popover-title');

            titleComponent.simulate(MOUSEOVER_EVENT);

            expect(Node.prototype.addEventListener).toHaveBeenCalledWith(MOUSEMOVE_EVENT, jasmine.anything());

            titleComponent.simulate(MOUSELEAVE_EVENT);

            return new Promise(resolve => {
              setTimeout(() => {
                resolve();
              }, DEFAULT_DELAY);
            }).then(() => {
              expect(Node.prototype.removeEventListener).toHaveBeenCalledWith(MOUSEMOVE_EVENT, jasmine.anything());
            });
          });
        });
      });
    });
  });
});
