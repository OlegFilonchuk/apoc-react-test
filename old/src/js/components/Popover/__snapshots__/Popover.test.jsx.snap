// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> default bevahior should have default title and placement = LEFT_TOP 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Some content for popover
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default placement and React.Element as a title should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <Button />
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default placement and simple text should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Simple Text Title
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="bottomLeft" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-bl qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="bottomMiddle" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-bm qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="bottomRight" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-br qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="leftMiddle" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lm qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="leftTop" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="rightMiddle" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-rm qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="rightTop" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-rt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="topLeft" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-tl qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="topMiddle" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-tm qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> rendering with default title and placement="topRight" should match snapshot 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-tr qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        >
          Simple Popover Text Content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Popover /> when \`popoverStyles\` property is specified should apply custom styles 1`] = `
<div
  className="tooltip-container "
  onClick={[Function]}
>
  <div
    className="popover-title"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Hover to see a tooltip
  </div>
  <div
    className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "border": "1px solid blue",
        "color": "green",
        "display": "none",
      }
    }
  >
    <div
      className="qtip-content"
      onClick={[Function]}
    >
      <div
        className="visible"
      >
        <header />
        <div
          className="tooltip-body"
        />
      </div>
    </div>
  </div>
</div>
`;
